### PR TITLE
Stackrox-ibmcloudz-qa-e2e-tests

### PR DESCRIPTION
**Pull request by gbhure**
Configurations to create/destroy k8s cluster in IBM Cloud for Z.
The required secret ibmcloudz-rhr-creds which is already added to the vault with IBM Cloud details

### CODE REVIEW

**Comment by tommartensen** [comment]
/cc @davdhacs @BradLugo 

@gbhure can you add some description to your PR: Jira issue, what & why?
Also please remove unnecessary debug output & check that your files have trailing empty lines.

**Comment by gbhure** [comment]
@tommartensen
I have updated description. Kindly review

File: ci-operator/step-registry/stackrox/automation-flavors/ibmcloudz-e2e/cluster-create/stackrox-automation-flavors-ibmcloudz-e2e-cluster-create-commands.sh [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+SSH_KEY_PATH="/tmp/id_rsa"
+SSH_ARGS="-i ${SSH_KEY_PATH} -o MACs=hmac-sha2-256 -o StrictHostKeyChecking=no -o LogLevel=ERROR -o UserKnownHostsFile=/dev/null"
+SSH_CMD="/home/ubuntu/stackrox-ci/k8smanager create"
+
+# setup ssh key
+cp -f $PRIVATE_KEY_FILE $SSH_KEY_PATH
+chmod 400 $SSH_KEY_PATH
+
+# create a k8s cluster
+ssh $SSH_ARGS root@163.66.94.115 "$SSH_CMD" > $KUBECONFIG_FILE
```
Comment (tommartensen) [inline ci-operator/step-registry/stackrox/automation-flavors/ibmcloudz-e2e/cluster-create/stackrox-automation-flavors-ibmcloudz-e2e-cluster-create-commands.sh:28]: This line is confusing to me: 
* who is `163.66.94.115`?
* I also need to find the `SSH_CMD` definition to understand what happens - creating a K8s cluster is not the first thing I'd associate with issuing a K8s command.

File: ci-operator/step-registry/stackrox/automation-flavors/ibmcloudz-e2e/cluster-destroy/stackrox-automation-flavors-ibmcloudz-e2e-cluster-destroy-commands.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+chmod +x kubectl
+
+SECRET_DIR="/tmp/vault/ibmcloudz-rhr-creds"
+PRIVATE_KEY_FILE="${SECRET_DIR}/PRIVATE_KEY_FILE"
+
+# https://docs.ci.openshift.org/docs/architecture/step-registry/#sharing-data-between-steps
+KUBECONFIG_FILE="${SHARED_DIR}/kubeconfig"
+
+# CLUSTER_NAME=$(KUBECONFIG="$KUBECONFIG_FILE" ./kubectl config view --minify -o jsonpath='{.clusters[].name}')
```
Comment (tommartensen) [inline ci-operator/step-registry/stackrox/automation-flavors/ibmcloudz-e2e/cluster-destroy/stackrox-automation-flavors-ibmcloudz-e2e-cluster-destroy-commands.sh:19]: Still required?

File: ci-operator/step-registry/stackrox/automation-flavors/ibmcloudz-e2e/cluster-create/stackrox-automation-flavors-ibmcloudz-e2e-cluster-create-commands.sh (hunk 1 above)
Comment (levivic) [reply ci-operator/step-registry/stackrox/automation-flavors/ibmcloudz-e2e/cluster-create/stackrox-automation-flavors-ibmcloudz-e2e-cluster-create-commands.sh:28]: hi @tommartensen, the goal here is for ibm z plaftform enablement for stackrox ci, i.e., to create and destroy ibm z K8s cluster on IBM Cloud as the first step. IBM Cloud currently doesn't support k8s z cluster management by default. So we have to remotely log into an intermediate node (163.66.94.115), and perform a installation/destroy through script called `k8smanager`, which is already set up there. This is a very similar process to the IBM power platform enablement: https://github.com/openshift/release/commit/498265a7f38b7bdc0420a9aec0331f438555cfb9

**Comment by gavin-stackrox** [comment]
@gbhure please `/cc @gavin-stackrox` when you open the new PR, that way openshift/release won't auto-assign people at random. thanks.

**Pull request closed without merging**